samples/matter/common/src/diagnostic/diagnostic_logs_provider.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#ifdef CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS
+
+#include "persistent_storage/persistent_storage_util.h"
+#include "util/finite_map.h"
+
+#include <app/clusters/diagnostic-logs-server/DiagnosticLogsProviderDelegate.h>
+#include <lib/core/CHIPError.h>
+
+namespace Nrf::Matter
+{
+
+class DiagnosticLogProvider : public chip::app::Clusters::DiagnosticLogs::DiagnosticLogsProviderDelegate {
+public:
+	static inline DiagnosticLogProvider &GetInstance() { return sInstance; }
+
+	CHIP_ERROR GetCrashLogs(chip::MutableByteSpan &outBuffer, bool &outIsEndOfLog);
+	CHIP_ERROR ClearCrashLogs();
```

Comment:
Are these 2 methods going to be used outside of this class? If no, we can make them private.